tests/test_Dataset.py

```diff
@@ -962,6 +962,60 @@ def _get_sub_epoch_dataset(files_subepoch: List[str]) -> Dict[str, Any]:
     assert global_seq_idx == num_hdf_files * num_seqs
 
 
+def test_PostprocessingDataset():
+    from returnn.tensor.tensor_dict import TensorDict
+
+    _demo_txt = "some utterance text that has a few words"
+
+    def _add_1337_to_classes(tdict: TensorDict) -> TensorDict:
+        tdict.data["classes"] += 1337
+        return tdict
+
+    with create_ogg_zip_txt_only_dataset_opts(text=_demo_txt) as sub_ds_opts:
+        ds_opts = {
+            "class": "PostprocessingDataset",
+            "dataset": sub_ds_opts,
+            "map_seq": _add_1337_to_classes,
+        }
+        dataset = init_dataset(ds_opts)
+        dataset.init_seq_order(epoch=1)
+        assert dataset.have_seqs()
+        dataset.init_seq_order(epoch=1)
+        dataset.load_seqs(0, 1)
+
+        classes = dataset.get_data(0, "classes")
+        assert all(c - 1337 >= 0 for c in classes)
+
+    i = 0
+
+    def _count(tdict: TensorDict) -> TensorDict:
+        nonlocal i
+        i += 1
+        return tdict
+
+    def _repeat2(input_iter: Iterator[TensorDict]) -> Iterator[TensorDict]:
+        for tdict in input_iter:
+            yield tdict
+            yield tdict
+
+    with create_ogg_zip_txt_only_dataset_opts(text=_demo_txt) as sub_ds_opts:
+        ds_opts = {
+            "class": "PostprocessingDataset",
+            "dataset": sub_ds_opts,
+            "map_seq": _count,
+            "map_seq_stream": _repeat2,
+        }
+        dataset = init_dataset(ds_opts)
+        dataset.init_seq_order(epoch=1)
+        assert dataset.have_seqs()
+        dataset.init_seq_order(epoch=1)
+
+        dataset.load_seqs(0, 2)
+        for i in range(2):
+            dataset.get_data(i, "classes")
+        assert i == 1
```

Comment:
I don't understand this check. Are you testing the Python stdlib `range`? You can assume that `range` is correct.

Reply:
Ah, I think I understand your intention now. You mix up the variable `i` here multiple times. You probably want a different name:
```suggestion
        assert count == 1
```